libnetwork/netutils/utils.go

```diff
@@ -121,14 +121,24 @@ func GenerateMACFromIP(ip net.IP) net.HardwareAddr {
 	return genMAC(ip)
 }
 
-// GenerateRandomName returns a new name joined with a prefix.  This size
-// specified is used to truncate the randomly generated value
-func GenerateRandomName(prefix string, size int) (string, error) {
-	id := make([]byte, 32)
-	if _, err := io.ReadFull(rand.Reader, id); err != nil {
+// GenerateRandomName returns a string of the specified length, created by joining the prefix to
+// random hex characters sufficient to create a string of the specified length.
+// If the specified length is less than or equal to the length of the prefix, an error will be
+// returned instead.
+func GenerateRandomName(prefix string, length uint8) (string, error) {
+	prefixLen := uint8(len(prefix))
+	if length <= prefixLen {
+		return "", fmt.Errorf("prefix %s is too long for length %d", prefix, length)
+	}
```

Comment:
As much as I like choosing types which can only represent in-range values, I don't think it helps prevent bugs to make `length` a small unsigned integer. The theoretical valid range for `length` is the interval

    (len(prefix), MaxInt-len(prefix)]

which always excludes zero, but every type in Go has a zero value so there is no way around having the runtime range check. And by the way, the runtime range check has a bug when `len(prefix) > 255`. The `len` builtin evaluates to `int` so it's an uphill battle to use any other type to represent lengths. Even `uint` is troublesome as it can represent a larger range of positive values than `int`.